### PR TITLE
Exclude endpoint routes list

### PR DESCRIPTION
**Pull request by wookieJ**

### CODE REVIEW

File: envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/listeners/filters/RBACFilterFactory.kt [hunk 1]
```diff
@@ -179,13 +179,18 @@ class RBACFilterFactory(
...
+                    rBACFilterPermissions.createPathPermission(
+                        path = it.path,
+                        matchingType = pathMatchingType
+                    ).build()
+                }
+                .asIterable()
             val policy = Policy.newBuilder()
                 .addPrincipals(anyPrincipal)
-                .addPermissions(permission)
+                .addPermissions(anyOf(permissions = permissions))
```
[the remark this answers is not in the document]
Comment (wookieJ) [reply envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/listeners/filters/RBACFilterFactory.kt]: removed

File: envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/listeners/filters/RBACFilterFactory.kt [hunk 2]
```diff
@@ -179,13 +179,18 @@ class RBACFilterFactory(
...
-            )
+            val permissions = statusRouteProperties.endpoints
+                .map {
+                    val pathMatchingType = PathMatchingType.valueOf(it.matchingType)
+                    rBACFilterPermissions.createPathPermission(
+                        path = it.path,
+                        matchingType = pathMatchingType
+                    ).build()
+                }
+                .asIterable()
```
[the remark this answers is not in the document]
Comment (wookieJ) [reply envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/listeners/filters/RBACFilterFactory.kt]: removed

File: envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/routes/EnvoyIngressRoutesFactory.kt [hunk 3]
```diff
@@ -34,13 +35,13 @@ class EnvoyIngressRoutesFactory(
...
-    private val statusPathPrefix = properties.routes.status.pathPrefix
+    private val statusEndpointsMatch: List<EndpointMatch> = properties.routes.status.endpoints
 
     private val endpointsPrefixMatcher = HeaderMatcher.newBuilder()
             .setName(":path").setPrefixMatch("/").build()
 
-    private val statusPrefixMatcher = HeaderMatcher.newBuilder()
-            .setName(":path").setPrefixMatch(statusPathPrefix).build()
+    private val statusPrefixMatcher: List<HeaderMatcher> = statusEndpointsMatch.map { HeaderMatcher.newBuilder()
+        .setName(":path").setPrefixMatch(it.path).build() }
```
[the remark this answers is not in the document]
Comment (wookieJ) [reply envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/routes/EnvoyIngressRoutesFactory.kt]: resolved

File: envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/routes/EnvoyIngressRoutesFactory.kt [hunk 4]
```diff
@@ -107,16 +108,20 @@ class EnvoyIngressRoutesFactory(
...
-                        .addHeaders(statusPrefixMatcher)
-                        .setName("status")
-                        .build(),
+                val endpoints = listOf(
                     VirtualCluster.newBuilder()
                         .addHeaders(endpointsPrefixMatcher)
                         .setName("endpoints")
                         .build()
                 )
+                val statusRoutes = statusPrefixMatcher
```
[the remark this answers is not in the document]
Comment (wookieJ) [reply envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/routes/EnvoyIngressRoutesFactory.kt]: moved to class fields and changed name `statusRoutes` to `statusClusters`

File: envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/listeners/filters/RBACFilterFactory.kt [hunk 5]
```diff
@@ -179,13 +179,18 @@ class RBACFilterFactory(
 
     private fun createStatusRoutePolicy(statusRouteProperties: StatusRouteProperties): Map<String, Policy> {
         return if (statusRouteProperties.enabled) {
-            val permission = rBACFilterPermissions.createPathPermission(
-                path = statusRouteProperties.pathPrefix,
-                matchingType = PathMatchingType.PATH_PREFIX
-            )
+            val permissions = statusRouteProperties.endpoints
+                .map {
+                    val pathMatchingType = PathMatchingType.valueOf(it.matchingType)
```
[the remark this answers is not in the document]
Comment (wookieJ) [reply envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/listeners/filters/RBACFilterFactory.kt]: Changed property type to `PathMatchingType`

File: envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/routes/EnvoyIngressRoutesFactory.kt [hunk 6]
```diff
@@ -34,13 +36,33 @@ class EnvoyIngressRoutesFactory(
...
 
-    private val statusPathPrefix = properties.routes.status.pathPrefix
+    private val statusEndpointsMatch: List<EndpointMatch> = properties.routes.status.endpoints
 
     private val endpointsPrefixMatcher = HeaderMatcher.newBuilder()
             .setName(":path").setPrefixMatch("/").build()
 
-    private val statusPrefixMatcher = HeaderMatcher.newBuilder()
-            .setName(":path").setPrefixMatch(statusPathPrefix).build()
+    private val statusPrefixMatcher: List<HeaderMatcher> = statusEndpointsMatch.map {
```
Comment (MarcinFalkowski) [inline envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/routes/EnvoyIngressRoutesFactory.kt:44]: The name `statusPrefixMatcher` is no longer accurate. I would change it to `statusMatcher`

File: envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/routes/EnvoyIngressRoutesFactory.kt [hunk 7]
```diff
@@ -40,8 +41,28 @@ class EnvoyIngressRoutesFactory(
     private val endpointsPrefixMatcher = HeaderMatcher.newBuilder()
             .setName(":path").setPrefixMatch("/").build()
 
-    private val statusPrefixMatcher: List<HeaderMatcher> = statusEndpointsMatch.map { HeaderMatcher.newBuilder()
-        .setName(":path").setPrefixMatch(it.path).build() }
+    private val statusPrefixMatcher: List<HeaderMatcher> = statusEndpointsMatch.map {
+        if (it.matchingType == PathMatchingType.PATH_PREFIX) {
```
[the remark this answers is not in the document]
Comment (wookieJ) [reply envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/routes/EnvoyIngressRoutesFactory.kt]: done

File: envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/routes/EnvoyIngressRoutesFactory.kt (hunk 6 above)
Comment (wookieJ) [reply envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/routes/EnvoyIngressRoutesFactory.kt]: done

File: envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/permissions/TlsBasedAuthenticationTest.kt [hunk 8]
```diff
@@ -22,7 +23,11 @@ internal class TlsBasedAuthenticationTest : EnvoyControlTestConfiguration() {
             "envoy-control.envoy.snapshot.outgoing-permissions.services-allowed-to-use-wildcard" to setOf("echo"),
             "envoy-control.envoy.snapshot.routes.status.create-virtual-cluster" to true,
             "envoy-control.envoy.snapshot.routes.status.endpoints" to mutableListOf(EndpointMatch().also { it.path = "/status/" }),
-            "envoy-control.envoy.snapshot.routes.status.enabled" to true
+            "envoy-control.envoy.snapshot.routes.status.enabled" to true,
+            "envoy-control.envoy.snapshot.routes.status.path-prefix" to "/status/",
+            "envoy-control.envoy.snapshot.routes.status.enabled" to true,
```
Comment (slonka) [inline envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/permissions/TlsBasedAuthenticationTest.kt:28]: Why twice?

Comment (wookieJ) [reply envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/permissions/TlsBasedAuthenticationTest.kt]: removed

**Pull request merged**